### PR TITLE
Feature/user module

### PR DESCRIPTION
**Pull request by jkao1**

### CODE REVIEW

File: src/js/modules/RoutingApp.js [hunk 1]
```diff
@@ -1,25 +1,28 @@
...
-const {ArticlePage} = articles.components;
-const {SectionPage} = sections.components;
+const { HomePage, PageLayout } = core.components;
+const { ArticlePage } = articles.components;
+const { SectionPage } = sections.components;
+const { RolePage, UserPage } = users.components;
 
 // TODO: change to mapStateToProps
 const allSectionRoutes = sections.selectors.getAllSectionRoutes(store.getState());
+const allRoles = users.selectors.getAllRoles(store.getState());
```
Comment (NicholasLYang) [inline src/js/modules/RoutingApp.js]: Why did you add another direct store use? There's literally a TODO there...

Comment (jkao1) [inline src/js/modules/RoutingApp.js]: If I connect() a mapStateToProps, I get `Invariant Violation: Could not find "store" in either the context or props of "Connect(RoutingApp)". Either wrap the root component in a <Provider>, or explicitly pass "store" as a prop to "Connect(RoutingApp)".`

File: src/js/modules/RoutingApp.js [hunk 2]
```diff
@@ -46,19 +45,51 @@ const RoutingApp = () => {
...
           </Switch>
         </PageLayout>
       </ConnectedRouter>
-    </Provider>
   );
 };
 
-export default RoutingApp;
+const mapStateToProps = (state) => ({
+  allSectionRoutes: sections.selectors.getAllSectionRoutes(state),
```
Comment (NicholasLYang) [inline src/js/modules/RoutingApp.js]: allSectionRoutes is weird. just do sectionRoutes

Comment (jkao1) [inline src/js/modules/RoutingApp.js]: changed in the refactor. I'll PR when i get to whole foodss

File: src/js/modules/articles/components/ArticlePage.js [hunk 3]
```diff
@@ -15,22 +13,26 @@ const styles = {
...
       />
       <ArticleBody content={article.content} featuredMedia={featuredMedia}/>
     </div>
   );
 };
 
 const mapStateToProps = (state, ownProps) => ({
-  article: getArticleBySlug(state, ownProps),
+  article: getArticleFromSlug(state, ownProps),
+  contributors: getContributorsOfArticle(state, ownProps),
```
Comment (NicholasLYang) [inline src/js/modules/articles/components/ArticlePage.js]: getArticleContributors

File: src/js/modules/articles/selectors.js [hunk 4]
```diff
@@ -1,26 +1,92 @@
...
-      return sectionAndSubsectionSlugs.includes(article.sectionSlug);
+      return slugsInSectionTree.includes(article.sectionSlug);
     });
   }
-);
\ No newline at end of file
+);
+
+export const getProcessedArticleResponseData = (state) => {
+  const data = state.articles.response;
```
Comment (NicholasLYang) [inline src/js/modules/articles/selectors.js]: data is a cool robot from star trek. Not a good variable name

File: dist/dev/css/main-index.css [hunk 5]
```diff
@@ -5,4 +5,87 @@
 
 body {
     margin: 0;
+}
+
+.col-lg-1, .col-lg-2, .col-lg-3, .col-lg-4, .col-lg-5, .col-lg-6, .col-lg-7, .col-lg-8, .col-lg-9, .col-lg-10, .col-lg-11, .col-lg-12,
```
Comment (NicholasLYang) [inline dist/dev/css/main-index.css]: 80 col rule

File: src/js/modules/RoutingApp.js [hunk 6]
```diff
@@ -1,64 +1,75 @@
...
+const RoutingApp = ({ sections, roles }) => {
   const createSectionRoutes = () => {
-    return Object.keys(allSectionRoutes).map((key, index) => {
-      const sectionRoute = allSectionRoutes[key];
+    return Object.keys(sections).map((sectionSlug, index) => {
+      const section = sections[ sectionSlug ];
       return <Route
-        exact path={sectionRoute.pathToSectionPage}
+        exact path={section.permalink}
         key={`sectionRoute${index}`}
```
Comment (NicholasLYang) [inline src/js/modules/RoutingApp.js]: Use sectionId

File: src/js/modules/RoutingApp.js [hunk 7]
```diff
@@ -1,64 +1,75 @@
...
   };
   const createArticleRoutes = () => {
-    return Object.keys(allSectionRoutes).map((key, index) => {
-      const sectionRoute = allSectionRoutes[key];
+    return Object.keys(sections).map((sectionSlug, index) => {
+      const section = sections[ sectionSlug ];
       return <Route
-        exact path={sectionRoute.pathToSectionPage + "/:article_slug"}
+        exact path={`${section.permalink}/:article_slug`}
         key={`articleRoute${index}`}
```
Comment (NicholasLYang) [inline src/js/modules/RoutingApp.js]: sectionId

File: src/js/modules/RoutingApp.js [hunk 8]
```diff
@@ -1,64 +1,75 @@
...
+    <ConnectedRouter history={appHistory}>
+      <PageLayout>
+        <Switch>
+          <Route exact path="/" component={HomePage}/>
+          {createSectionRoutes()}
+          {createArticleRoutes()}
+          {createRoleRoutes()}
+          <Route exact path={'/contributors/:contributor_slug'}
+                 key={`contributorRoute`}
+                 render={props => <ContributorPage match={props.match} role={roles[ 'contributors' ]}/>}/>
```
Comment (NicholasLYang) [inline src/js/modules/RoutingApp.js]: 80 column rule

File: src/js/modules/articles/actions.js [hunk 9]
```diff
@@ -1,11 +1,72 @@
-import axios from 'axios';
-import * as t from './actionTypes';
+import axios from "axios";
+import * as t from "./actionTypes";
+import { STUY_SPEC_API, HEADER } from "../../constants";
+import { getProcessedArticlesResponse, getFakeAuthorshipsForArticleResponse } from "./selectors";
 
-const apiUrl = 'http://api.icndb.com/jokes/random';
+export const fetchArticles = () => {
+  return (dispatch, getState) => {
```
Comment (NicholasLYang) [inline src/js/modules/articles/actions.js]: do you need the getState?

File: src/js/modules/articles/actions.js [hunk 10]
```diff
@@ -1,11 +1,72 @@
...
+    axios.get(`${STUY_SPEC_API}/articles`, { 'headers': HEADER })
+      .then(response => {
+        if (areArticlesValid(response.data)) {
+          dispatch({
+            type: t.FETCH_ARTICLE_FULFILLED,
+            payload: response.data,
+          });
+        }
+      })
+      .then(response => { // TODO: promise function orders are wonky without @param response
```
Comment (NicholasLYang) [inline src/js/modules/articles/actions.js]: Why is this a separate .then?

File: src/js/modules/articles/actions.js [hunk 11]
```diff
@@ -1,11 +1,72 @@
...
+      })
+  };
+};
 
-export const fetch = () => {
-	return {
-		type: t.FETCH_ARTICLE,
-		payload: axios.get(apiUrl),
-	};
+const checkArticleKeyValidity = (articleObject, key, type) => {
```
Comment (NicholasLYang) [inline src/js/modules/articles/actions.js]: Awkward. validateArticleKey makes more sense

File: src/js/modules/articles/actions.js [hunk 12]
```diff
@@ -1,11 +1,72 @@
...
+import { STUY_SPEC_API, HEADER } from "../../constants";
+import { getProcessedArticlesResponse, getFakeAuthorshipsForArticleResponse } from "./selectors";
 
-const apiUrl = 'http://api.icndb.com/jokes/random';
+export const fetchArticles = () => {
+  return (dispatch, getState) => {
+    dispatch({ type: t.FETCH_ARTICLE_PENDING });
+    axios.get(`${STUY_SPEC_API}/articles`, { 'headers': HEADER })
+      .then(response => {
+        if (areArticlesValid(response.data)) {
```
Comment (NicholasLYang) [inline src/js/modules/articles/actions.js]: validateArticles

File: src/js/modules/articles/selectors.js [hunk 13]
```diff
@@ -1,26 +1,140 @@
...
+      });
+    }
+  );
+};
+
+/**
+ * The selector factory returns a selector that returns the byline for
+ *   any @param targetArticle.
+ */
+export const articleBylineSelectorFactory = (targetArticle) => {
```
Comment (NicholasLYang) [inline src/js/modules/articles/selectors.js]: Why????? Batch process bylines for every article

File: dist/dev/css/main-index.css (hunk 5 above)
Comment (jkao1) [inline dist/dev/css/main-index.css]: it'd be less readable if i split up the lines

File: src/js/modules/RoutingApp.js (hunk 6 above)
Comment (jkao1) [inline src/js/modules/RoutingApp.js]: done

File: src/js/modules/RoutingApp.js (hunk 7 above)
Comment (jkao1) [inline src/js/modules/RoutingApp.js]: done

File: src/js/modules/RoutingApp.js (hunk 8 above)
Comment (jkao1) [inline src/js/modules/RoutingApp.js]: done

File: src/js/modules/articles/actions.js (hunk 9 above)
Comment (jkao1) [inline src/js/modules/articles/actions.js]: yes. look in dispatches

File: src/js/modules/articles/actions.js (hunk 12 above)
Comment (jkao1) [inline src/js/modules/articles/actions.js]: done

File: src/js/modules/articles/actions.js (hunk 10 above)
Comment (jkao1) [inline src/js/modules/articles/actions.js]: done

File: src/js/modules/articles/actions.js (hunk 11 above)
Comment (jkao1) [inline src/js/modules/articles/actions.js]: yup

File: dist/dev/css/main-index.css (hunk 5 above)
Comment (NicholasLYang) [inline dist/dev/css/main-index.css]: No, it really won't be

Comment (NicholasLYang) [inline dist/dev/css/main-index.css]: Also, why are you hard coding your cols? I'm not a CSS expert here, but why does this need to be done in CSS and not JSS?

File: src/js/modules/articles/actions.js (hunk 12 above)
Comment (NicholasLYang) [inline src/js/modules/articles/actions.js]: push it then

File: src/js/modules/articles/actions.js [hunk 14]
```diff
@@ -1,11 +1,69 @@
-import axios from 'axios';
-import * as t from './actionTypes';
+import axios from "axios";
+import * as t from "./actionTypes";
+import { STUY_SPEC_API, HEADER } from "../../constants";
```
Comment (NicholasLYang) [inline src/js/modules/articles/actions.js]: Super minor, but consider making HEADER more descriptive. Like STUY_API_HEADER

File: src/js/modules/articles/selectors.js [hunk 15]
```diff
@@ -1,26 +1,127 @@
...
-const getSectionByProps = (state, props) => props.section;
 
-export const getArticleBySlug = createSelector(
-  [ getSectionByProps, getArticles, getRequestedArticleSlug ],
-  (section, articles, articleSlug) => {
-    const requestedArticle = articles[ articleSlug ];
-    // the article may match the slug, but does it match the requested section?
+/**
+ * The selector returns an articles object with an additional property for each
+ *   article: its byline in JSX.
```
Comment (NicholasLYang) [inline src/js/modules/articles/selectors.js]: Update comment

File: src/js/modules/articles/selectors.js [hunk 16]
```diff
@@ -1,26 +1,127 @@
...
+ */
+const getArticlesWithContributors = createSelector(
+  [ getArticles, getUsers, getAuthorships ],
+  (articles, users, authorships) => {
+    let articlesWithContributors = articles;
```
Comment (NicholasLYang) [inline src/js/modules/articles/selectors.js]: This is a little messy, I don't like the undefined check. How about you loop through articles, then filter the authorships according to article_slug, map them over users and then add them to the articles object

Comment (NicholasLYang) [inline src/js/modules/articles/selectors.js]: You don't have to do that, but you do need more intermediary variables

Comment (jkao1) [inline src/js/modules/articles/selectors.js]: I looped through authorships in the interest of efficiency. If i looped through users or articles, I would be going through all authorships several times.

File: src/js/modules/articles/selectors.js [hunk 17]
```diff
@@ -1,26 +1,127 @@
...
+/**
+ * The selector returns an articles object with an additional property for each
+ *   article: an array of its contributors.
+ */
+const getArticlesWithContributors = createSelector(
+  [ getArticles, getUsers, getAuthorships ],
+  (articles, users, authorships) => {
+    let articlesWithContributors = articles;
+    authorships.map(authorship => {
+      if (articlesWithContributors[ authorship.articleSlug ].contributors === undefined) {
```
Comment (NicholasLYang) [inline src/js/modules/articles/selectors.js]: This needs an intermediary variable

File: src/js/modules/articles/selectors.js [hunk 18]
```diff
@@ -1,26 +1,127 @@
...
+        articlesWithContributors[ authorship.articleSlug ].contributors = [
+          users[ authorship.contributorSlug ],
+        ];
+      } else {
+        articlesWithContributors[ authorship.articleSlug ].contributors
+          .push(users[ authorship.contributorSlug ]);
+      }
+    });
+    Object.keys(articlesWithContributors).map(articleSlug => {
```
Comment (NicholasLYang) [inline src/js/modules/articles/selectors.js]: This is validation. Not sure it should be in the selector

File: src/js/modules/articles/selectors.js [hunk 19]
```diff
@@ -1,26 +1,127 @@
...
-    return Object.filter(articles, article => {
-      return sectionAndSubsectionSlugs.includes(article.sectionSlug);
+/**
+ * The selector returns a filtered articles object that contains all articles
+ *   written by a contributor.
+ */
+export const getArticlesByContributor = createSelector(
+  [ getContributorFromSlug, getArticlesWithContributors, getAuthorships ],
+  (contributor, articlesWithContributors, authorships) => {
+    const articleSlugsForArticlesByContributor = authorships
```
Comment (NicholasLYang) [inline src/js/modules/articles/selectors.js]: ...honestly would just call it articleSlugs

File: src/js/modules/core/components/HomePage.js [hunk 20]
```diff
@@ -3,57 +3,78 @@ import { bindActionCreators } from "redux";
...
-  const createSectionListItems = () => {
-    return Object.keys(sectionsWithLinks).map((key) => {
+const HomePage = ({ classes, sections, articles, users, userRoles, fetchArticles, fetchUsers }) => {
+  const linkToAllArticles = () => {
+    return Object.keys(articles).map((articleSlug, index) => {
+      const article = articles[ articleSlug ];
       return (
-        <li key={key}>
-          <Link to={sectionsWithLinks[ key ].pathToSectionPage}>{sectionsWithLinks[ key ].name}</Link>
+        <li key={`articleLink${index}`}>
```
Comment (NicholasLYang) [inline src/js/modules/core/components/HomePage.js]: Avoid using index

File: src/js/modules/core/components/HomePage.js [hunk 21]
```diff
@@ -3,57 +3,78 @@ import { bindActionCreators } from "redux";
...
 
 const styles = {
   HomePage: {}
 };
 
-const HomePage = ({ classes, sectionsWithLinks, articles }) => {
-  const createSectionListItems = () => {
-    return Object.keys(sectionsWithLinks).map((key) => {
+const HomePage = ({ classes, sections, articles, users, userRoles, fetchArticles, fetchUsers }) => {
+  const linkToAllArticles = () => {
```
Comment (NicholasLYang) [inline src/js/modules/core/components/HomePage.js]: I liked create for generating links. Or even generateAllArticleLinks

File: src/js/modules/core/components/MastheadBar.js [hunk 22]
```diff
@@ -34,22 +33,58 @@ const styles = {
...
+  buttonText: {
+    color: '#000',
+    fontFamily: 'Circular Std',
+    fontSize: '12px',
+    fontWeight: 'bold',
+    textTransform: 'uppercase',
   }
 };
 
+const UnstyledNavButton = ({ classes, label }) => {
```
Comment (NicholasLYang) [inline src/js/modules/core/components/MastheadBar.js]: little awkward. I'd just make a separate file or just call the second one StyledNavButton

File: src/js/modules/core/components/PageFooter.js [hunk 23]
```diff
@@ -23,78 +23,83 @@ const styles = {
...
-
   return (
     <Grid className={classes.PageFooter}>
       <Row className={classes.pageFooterMain}>
         <Col md={8} mdOffset={2} className={classes.theSpectator}>
           The Spectator
         </Col>
         <Col md={8} mdOffset={2} className={classes.sectionFlex}>
-          {makeSectionLinks()}
+          {linkToSections()}
```
Comment (NicholasLYang) [inline src/js/modules/core/components/PageFooter.js]: linkToSections is a noun. You want a verb

File: src/js/modules/sections/components/SectionPage.js [hunk 24]
```diff
@@ -1,75 +1,96 @@
...
-  const createSubsectionLinks = () => {
-    return Object.keys(subsections).map((key) => {
-      const subsection = subsections[ key ];
+const SectionPage = ({ classes,
+                       sectionTreeArticles,
+                       directSubsectionChildren,
+                       section,
+                       featuredMedia,
+                     }) => {
+  const createLinksToDirectSubsectionChildren = () => {
```
Comment (NicholasLYang) [inline src/js/modules/sections/components/SectionPage.js]: DirectSubsectionChildren is wordy and a little weird. Do you mean the direct children of subsections? The direct children that are subsections?

File: src/js/modules/sections/components/SectionPage.js [hunk 25]
```diff
@@ -1,75 +1,96 @@
...
-        {createSubsectionLinks()}
-      </ul>
-      <hr/>
-      <p>articles</p>
-      <ul>
-        {createArticleLinks()}
-      </ul>
+      <h1 className={classes.sectionName}>{section.name}</h1>
+      {
+        directSubsectionChildren !== null &&
```
Comment (NicholasLYang) [inline src/js/modules/sections/components/SectionPage.js]: What about undefined?

File: src/js/modules/sections/components/SectionPage.js [hunk 26]
```diff
@@ -1,75 +1,96 @@
...
 const mapStateToProps = (state, ownProps) => ({
-  articles: articles.selectors.getArticlesWithinSection(state, ownProps),
+  featuredMedia: {
+    url: 'http://planesandpleasures.com/wp-content/uploads/2016/09/NewYork-Chinatown-7.jpg',
+    caption: 'New York City street after rain is covered in water, dirt, and snow. Pedestrians walk back and forth as post-flood confusion amasses.',
+    type: 'Photograph',
+    credits: 'Ting Ting',
+  },
+  sectionTreeArticles: getSectionTreeArticles(state, ownProps),
+  directSubsectionChildren: getDirectChildrenOfSection(state, ownProps),
```
Comment (NicholasLYang) [inline src/js/modules/sections/components/SectionPage.js]: getDirectChildrenOfSection is awkward

File: src/js/modules/sections/reducer.js [hunk 27]
```diff
@@ -16,132 +10,132 @@ const initialState = {
       slug: "news",
       description: "The news of Stuyvesant.",
       parentSlug: null,
+      permalink: "/news"
```
Comment (NicholasLYang) [inline src/js/modules/sections/reducer.js]: Not gonna make you change this, but why not have selectors create permalink?

File: src/js/modules/articles/selectors.js (hunk 18 above)
Comment (jkao1) [inline src/js/modules/articles/selectors.js]: Where else would it go? It's in the selector that it is processed.

File: src/js/modules/sections/components/SectionPage.js (hunk 24 above)
Comment (jkao1) [inline src/js/modules/sections/components/SectionPage.js]: yes

File: src/js/modules/sections/reducer.js (hunk 27 above)
Comment (jkao1) [inline src/js/modules/sections/reducer.js]: permalinks will be generated when we process response data from API. so, yes, it will be in the selectors in the future.

File: src/js/modules/sections/components/SectionPage.js (hunk 25 above)
Comment (jkao1) [inline src/js/modules/sections/components/SectionPage.js]: if any selector fails, it will return null not undefined.

File: src/js/modules/users/selectors.js [hunk 28]
```diff
@@ -0,0 +1,45 @@
...
+  [ getUsers, getRequestedContributorSlug ],
+  (users, requestedContributorSlug) => {
+    return users[ requestedContributorSlug ];
+  }
+);
+
+/**
+ * The selector returns all users within a role.
+ */
+export const getUsersWithinRole = createSelector(
```
Comment (NicholasLYang) [inline src/js/modules/users/selectors.js]: Little awkward. getUserByRole?

Comment (jkao1) [inline src/js/modules/users/selectors.js]: That function name incorrectly describes the function. But I'll rename it to getUsersInRole

File: src/js/modules/users/selectors.js [hunk 29]
```diff
@@ -0,0 +1,45 @@
...
+      .filter(userRole => userRole.roleSlug === role.slug)
+      .map(userRole => userRole.userSlug);
+    return Object.filter(users, user => {
+      return userSlugsInRole.includes(user.slug);
+    });
+  }
+);
+
+/**
+ * The selector returns an articles object that contains all articles from Stuy
```
Comment (NicholasLYang) [inline src/js/modules/users/selectors.js]: Users or articles

File: src/js/modules/users/selectors.js [hunk 30]
```diff
@@ -0,0 +1,45 @@
...
+
+export const getUsers = state => state.users.users;
+export const getRoles = state => state.users.roles;
+export const getUserRoles = state => state.users.userRoles;
+
+const getRequestedContributorSlug = (state, props) => props.match.params.contributor_slug;
+const getRoleFromProps = (state, props) => props.role;
+const getUsersResponse = state => state.articles.response;
+
+export const getContributorFromSlug = createSelector(
```
Comment (NicholasLYang) [inline src/js/modules/users/selectors.js]: Why not getRequestedContributor?

File: src/js/modules/users/selectors.js [hunk 31]
```diff
@@ -0,0 +1,45 @@
+import { createSelector } from "reselect";
+
+export const getUsers = state => state.users.users;
+export const getRoles = state => state.users.roles;
+export const getUserRoles = state => state.users.userRoles;
+
+const getRequestedContributorSlug = (state, props) => props.match.params.contributor_slug;
+const getRoleFromProps = (state, props) => props.role;
+const getUsersResponse = state => state.articles.response;
```
Comment (NicholasLYang) [inline src/js/modules/users/selectors.js]: users or articles?

File: src/js/modules/users/selectors.js (hunk 30 above)
Comment (jkao1) [inline src/js/modules/users/selectors.js]: Then i wouldn't know if that's from props or slug.

File: src/js/modules/users/selectors.js (hunk 31 above)
Comment (NicholasLYang) [inline src/js/modules/users/selectors.js]: Cause it says users but it's fetching from articles

File: src/js/modules/users/actions.js [hunk 32]
```diff
@@ -0,0 +1,62 @@
...
+    } else {
+      throw `EXCEPTION: key ${key} in userObject is 
+        ${typeof (userObject[ key ])}, but should be ${type}.`;
+    }
+  } else {
+    throw `EXCEPTION: key ${key} is undefined in userObject.`;
+  }
+};
+
+const areUsersValid = (userArray) => {
```
Comment (NicholasLYang) [inline src/js/modules/users/actions.js]: c'mon. keep it consistent. validateUsers

File: src/js/modules/users/actions.js [hunk 33]
```diff
@@ -0,0 +1,62 @@
...
+      .catch((err) => {
+        dispatch({
+          type: t.FETCH_USER_REJECTED,
+          payload: err,
+        })
+      })
+  };
+};
+
+const checkUserKeyValidity = (userObject, key, type) => {
```
Comment (NicholasLYang) [inline src/js/modules/users/actions.js]: validateUserKey

File: src/js/modules/users/actions.js [hunk 34]
```diff
@@ -0,0 +1,62 @@
...
+      .then(response => {
+        if (areUsersValid(response.data)) {
+          dispatch({
+            type: t.FETCH_USER_FULFILLED,
+            payload: response.data,
+          });
+        }
+      })
+      .then(response => {
+        dispatch({
```
Comment (NicholasLYang) [inline src/js/modules/users/actions.js]: You don't need a .then for every dispatch. Not a big deal, just saying

File: src/js/modules/sections/selectors.js [hunk 35]
```diff
@@ -1,108 +1,77 @@
...
+ *   indirect children of a target section.
  */
-export const getSectionAndSubsectionSlugs = createSelector(
-  [ getSectionByProps, getSubsectionsByProps ],
-  (section, subsections) => {
-    return [ section.slug, ...Object.keys(subsections) ];
-  }
-);
+const getSubsectionsInSectionTree = (sections, targetSection) => {
+  return Object.filter(sections, sectionTraversingUpwards => {
```
Comment (NicholasLYang) [inline src/js/modules/sections/selectors.js]: sectionTraversingUpwards is a confusing name

Comment (jkao1) [inline src/js/modules/sections/selectors.js]: If you look at the code, you'll see why I named it that. I thought it described the behavior well, but I admit that it doesn't describe the subsection well. Do you have a better name?

File: src/js/modules/users/selectors.js (hunk 28 above)
Comment (NicholasLYang) [inline src/js/modules/users/selectors.js]: Well it's getting all the users by the role you want

Comment (NicholasLYang) [inline src/js/modules/users/selectors.js]: But yeah getUsersInRole still works

File: src/js/modules/users/selectors.js (hunk 30 above)
Comment (NicholasLYang) [inline src/js/modules/users/selectors.js]: getRequestedContributorBySlug?
